synapse/util/threepids.py

```diff
@@ -38,10 +38,18 @@ def check_3pid_allowed(hs, medium, address):
         data = yield hs.get_simple_http_client().get_json(
             "https://%s%s" % (
                 hs.config.check_is_for_allowed_local_3pids,
-                "/_matrix/identity/api/v1/info"
+                "/_matrix/identity/api/v1/internal-info"
             ),
             {'medium': medium, 'address': address}
         )
+
+        # Assume false if invalid response
+        if 'hs' not in data:
+            defer.returnValue(False)
+
+        if data.get('requires_invite', False) and data.get('invited', False) == False:
```

Comment:
This is not very idiomatic, and is a bit confusing.

```suggestion
        if data.get('requires_invite', False) and not data.get('invited', False):
```